engine/servermaster/jobmanager.go

```diff
@@ -482,7 +486,7 @@ func (jm *JobManagerImpl) GetJobMasterForwardAddress(ctx context.Context, jobID
 		}
 		return "", err
 	}
-	if masterMeta.State != frameModel.MasterStateInit || jm.JobFsm.QueryOnlineJob(jobID) == nil {
+	if masterMeta.State != frameModel.MasterStateInit && jm.JobFsm.QueryOnlineJob(jobID) == nil {
```

Comment:
Why change the logic here?

Reply:
This's a bug. We shouldn't forward requests to the job master when the job is not online.
It's also the root cause of https://github.com/pingcap/tiflow/issues/7555.

Reply:
Confused, does `job online` mean
`masterMeta.State == frameModel.MasterStateInit && jm.JobFsm.QueryOnlineJob(jobID) != nil`
Seems the original check is correct

Reply:
Fixed.